client/src/modules/CrossCheck/components/criteria/SubtaskCriteria.tsx

```diff
@@ -1,76 +1,79 @@
-import { Input, Rate, Typography } from 'antd';
+import { Input, Typography, InputNumber, Slider } from 'antd';
+import { useMemo } from 'react';
 import { CrossCheckCriteriaData } from 'services/course';
 
-import { CommentState, CountState } from '../CrossCheckCriteriaForm';
-
 const { TextArea } = Input;
+const { Text } = Typography;
+
 interface SubtaskCriteriaProps {
-  criteriaComment: CommentState[];
   subtaskData: CrossCheckCriteriaData;
-  countStar: CountState[];
-  updateCountStar: (event: number, max: number, key: string) => void;
-  updateComment: (value: string, key: string) => void;
+  updateCriteriaData: (updatedEntry: CrossCheckCriteriaData) => void;
 }
 
-const NUMBER_OF_STARS = 5;
+export default function SubtaskCriteria({ subtaskData, updateCriteriaData }: SubtaskCriteriaProps) {
+  const maxScore = subtaskData.max as number;
+  const comment = subtaskData.textComment as string;
+  const criteriaScore = subtaskData.point as number;
+
+  const updateScore = (event: number) => {
+    const updatedEntry = { ...subtaskData, point: event };
+    updateCriteriaData(updatedEntry);
+  };
 
-export default function SubtaskCriteria({
-  criteriaComment,
-  subtaskData,
-  countStar,
-  updateCountStar,
-  updateComment,
-}: SubtaskCriteriaProps) {
-  const criteriaScore = (countStar.find(item => item.key === subtaskData.key)?.point as number) ?? 0;
+  const updateComment = (newComment: string) => {
+    const updatedEntry = { ...subtaskData, textComment: newComment };
+    updateCriteriaData(updatedEntry);
+  };
+
+  const statusCommentRequired = useMemo(() => {
+    if (criteriaScore !== undefined) {
```

Comment:
Here is the problem that I was talking above - you check value that is declared as `number` for not being `undefined`, this should be fixed by using proper types :)